### PR TITLE
CAPA v29.1.0: Define catalog for `cilium-crossplane-resources`.

### PR DESCRIPTION
**Pull request by Gacko**
This PR explicitly defines the catalog for the `cilium-crossplane-resources` app.

At the moment, `cluster-aws` is hard-coding this catalog anyways. So merging this PR will effectively not change any existing clusters running on this release.

But as I'm currently about to implement the same logic of looking up catalogs from the Release CR in the `cluster-aws` chart as we already have in the `cluster` chart, the Cluster Test Suites might fail as they are picking the latest available Release CR, patch the `cluster-aws` chart version to test and then execute their tests.

In this specific scenario `cilium-crossplane-resources` would be rendered with `catalog: default` as without this PR the latest Release CR does not explicitly define a catalog for it and therefore falls back to `default`.

**tl;dr:** This PR has no effect on production clusters, it is only required for development.

### Triggering E2E tests

To trigger the E2E test for each new Release added in this PR, add a comment with the following:

`/run releases-test-suites`

If you want to trigger conformance tests, you can do so by adding a comment similar to the following:

`/run conformance-tests PROVIDER=capa RELEASE_VERSION=v29.0.0`

For more details see the [README.md](/README.md#running-tests-against-prs).

### CODE REVIEW

**Comment by Gacko** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capa/standard,./providers/capa/upgrade PREVIOUS_RELEASE=28.1.1 TARGET_RELEASES=aws-29.1.0

**Review by fiunchinho — APPROVED** [review]
